ouroboros-consensus/docs/ChainSync.md

```diff
@@ -64,10 +99,23 @@ We get notified through the chain sync protocol about updates to their chain:
 
 Our fragment is always `k` long (unless near genesis), so no trimming needed.
 However, to avoid unbounded memory usage, we should trim their fragment.
-It should always be safe to trim to the intersection point.
 
-NOTE: This may leave their fragment shorter than `k`, thus not allowing them
-their full rollback. However, this is okay: if their fragment _is_ shorter than
-`k` due to trimming to the intersection point, and they _do_ roll back to a
-point before their fragment, this new intersection point would be too far from
-our tip, and we should disconnect from them.
+We have established that the intersection between their chain and our _chain_
+must lie on both _fragments_, and moreover, that this intersection will be
+within `k` from our tip; since our fragment is `k` long, it means the
+intersection will be somewhere along our fragment.
+
+This means that we should allow the remote node to roll back from its current
+intersection point to our chain to any point on our fragment, but we do not need
+to allow it to roll back more than that. This means we can always trim their
+fragment so that it is anchored at our anchor point.
+
+- In the common case that both nodes are following the same chain and both up
+  to date, this would leave fragments identical.
+- In the extreme case that they are `k` behind, it would leave their fragment
+  empty.
+- More generally, this may leave their fragment shorter than `k`, thus not
+  allowing them their full rollback. However, this is okay: if their fragment
+  _is_ shorter than `k` due to trimming to the intersection point, and they _do_
```

Comment:
intersection point -> anchor point?

Reply:
ah, yes, well spotted, that was a remnant from our previous attempt.